File: app/src/main/java/com/icebem/akt/ui/tools/ToolsFragment.java
package com.icebem.akt.ui.tools;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.icebem.akt.R;
import com.icebem.akt.activity.AboutActivity;
import com.icebem.akt.activity.MainActivity;
import com.icebem.akt.model.RecruitViewer;
import com.icebem.akt.util.AppUtil;

import java.util.ArrayList;

public class ToolsFragment extends Fragment {
    private RecruitViewer viewer;

    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        View root = inflater.inflate(R.layout.fragment_tools, container, false);
        try {
            viewer = new RecruitViewer(getActivity(), root);
        } catch (Exception e) {
            if (getActivity() instanceof MainActivity)
                AppUtil.showLogDialog(getActivity(), e);
        }
        return root;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_overlay:
                if (getActivity() instanceof MainActivity)
                    ((MainActivity) getActivity()).showOverlay();
                break;
            case R.id.action_language:
                if (viewer == null) break;
                ArrayList<String> packages = viewer.getManager().getAvailablePackages();
                int index = viewer.getManager().getGamePackagePosition();
                if (++index == packages.size()) index = 0;
                viewer.getManager().setGamePackage(packages.get(index));
                viewer.resetTags(null);
                break;
            case R.id.action_about:
                startActivity(new Intent(getActivity(), AboutActivity.class));
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        inflater.inflate(R.menu.menu_tools, menu);
        if (viewer != null && viewer.getManager().multiPackage())
            menu.findItem(R.id.action_language).setVisible(true);
    }
}

File: app/src/main/java/com/icebem/akt/ui/tools/ToolsFragment.kt
package com.icebem.akt.ui.tools

import android.content.Intent
import android.os.Bundle
import android.view.*
import androidx.core.view.MenuHost
import androidx.core.view.MenuProvider
import androidx.fragment.app.Fragment
import androidx.lifecycle.Lifecycle
import com.icebem.akt.R
import com.icebem.akt.activity.AboutActivity
import com.icebem.akt.activity.MainActivity
import com.icebem.akt.model.RecruitViewer
import com.icebem.akt.util.AppUtil

class ToolsFragment : Fragment(), MenuProvider {
    private var viewer: RecruitViewer? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val menuHost = requireActivity() as MenuHost
        menuHost.addMenuProvider(this, viewLifecycleOwner, Lifecycle.State.RESUMED)
        val root = inflater.inflate(R.layout.fragment_tools, container, false)
        try {
            viewer = RecruitViewer(requireContext(), root)
        } catch (e: Exception) {
            AppUtil.showLogDialog(requireContext(), e.toString())
        }
        return root
    }

    override fun onMenuItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_overlay -> (requireContext() as MainActivity).showOverlay()
            R.id.action_language -> {
                val packages = viewer!!.manager.availablePackages
                var index = viewer!!.manager.gamePackagePosition
                if (++index == packages.size) index = 0
                viewer!!.manager.setGamePackage(packages[index])
                viewer!!.resetTags(null)
            }
            R.id.action_about -> startActivity(Intent(requireContext(), AboutActivity::class.java))
        }
        return false
    }

    override fun onCreateMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.menu_tools, menu)
        if (viewer != null && viewer!!.manager.multiPackage) menu.findItem(R.id.action_language).isVisible = true
    }
}